src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -2149,4 +2149,58 @@ public void Reset()
         }
 #endif // !CORERT
     }
+
+    internal sealed class SZGenericArrayEnumerator<T> : IEnumerator<T>
```

Comment:
It was nested class before in CoreCLR. I would keep it as nested class.

Also, I would keep it under the `!CORERT` ifdef above. It won't be usable in CoreRT just yet.

Reply:
> keep it as nested class

I guess you do not want to take the SZArrayHelper with this.

Should it rather be just a separate file?

Reply:
yeah, moved all enumerators to a new file